qiskit/result/utils.py

```diff
@@ -35,8 +41,8 @@ def marginal_counts(result, indices=None):
     Raises:
         QiskitError: in case of invalid indices to marginalize over.
     """
-    from qiskit.result.result import Result
-    from qiskit.result.postprocess import _bin_to_hex
+    if not in_place:
+        result = deepcopy(result)
```

Comment:
If I understand correctly, the bug can only happens in the else-branch of the coming conditional. Would it be possible to only make the deepcopy in that case?

Reply:
Sure. Looking into it!

Reply:
The error occurs because the original job object is still being referenced. It seems regardless of which condition turns out to be true, the bug occurs. Which is why deep copying the result argument before the invocation of ```_marginalize()``` at lines 47 and 57 of result/utils.py solves the problem.